### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Daniil-Pankieiev**

### CODE REVIEW

**Review by petrykivd — APPROVED** [review]

File: app/main.py [hunk 1]
```diff
@@ -1,214 +1,35 @@
...
+        all_knights.append(
+            Knight(
+                knight["name"],
+                knight["power"],
+                knight["hp"],
+                knight["armour"],
+                knight["weapon"],
+                knight["potion"]
+            )
+        )
```
Comment (LiudmylaKulinchenko) [inline app/main.py:19]: Use unpacking `**` and list comprehension

File: app/main.py [hunk 2]
```diff
@@ -1,214 +1,35 @@
...
-            red_knight["power"] += red_knight["potion"]["effect"]["power"]
-
-        if "protection" in red_knight["potion"]["effect"]:
-            red_knight["protection"] += red_knight["potion"]["effect"]["protection"]
-
-        if "hp" in red_knight["potion"]["effect"]:
-            red_knight["hp"] += red_knight["potion"]["effect"]["hp"]
-
-    # -------------------------------------------------------------------------------
+    # Create a list of instances of the Knight class
```
Comment (LiudmylaKulinchenko) [inline app/main.py:7]: You can remove comments, the logic is clear

File: app/preparations_for_batle/apply_equipment.py [hunk 3]
```diff
@@ -0,0 +1,18 @@
+class PrepareForBattle:
+    def apply_weapon(self) -> None:
+        self.power += self.weapon["power"]
+
+    def apply_armour(self) -> None:
+        for armour in self.armour:
+            self.protection += armour["protection"]
```
Comment (LiudmylaKulinchenko) [inline app/preparations_for_batle/apply_equipment.py:7]: U can use sum here

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/preparations_for_batle/apply_equipment.py [hunk 4]
```diff
@@ -3,8 +3,7 @@ def apply_weapon(self) -> None:
         self.power += self.weapon["power"]
 
     def apply_armour(self) -> None:
-        for armour in self.armour:
-            self.protection += armour["protection"]
+        self.protection = sum([armour["protection"] for armour in self.armour])
```
Comment (LiudmylaKulinchenko) [inline app/preparations_for_batle/apply_equipment.py:6]: sum works with generators so [] are not needed

File: app/start_battle/act_battle.py [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+        second_knight.hp = BattleKnights.check_hp(second_knight)
+
+    @staticmethod
+    def battle_result(knights: list) -> dict:
+        return {
+            knights[0].name: knights[0].hp,
+            knights[1].name: knights[1].hp,
+            knights[2].name: knights[2].hp,
+            knights[3].name: knights[3].hp,
+        }
```
Comment (LiudmylaKulinchenko) [inline app/start_battle/act_battle.py:25]: You can use dict compehension

**Review by LiudmylaKulinchenko — CHANGES_REQUESTED** [review]

File: app/preparations_for_batle/apply_equipment.py (hunk 4 above)
Comment (Daniil-Pankieiev) [reply app/preparations_for_batle/apply_equipment.py]: Fixed

**Review by LiudmylaKulinchenko — APPROVED** [review]
Good job! 🌼🌺

**Pull request closed without merging**